Tournament.cpp

```diff
@@ -0,0 +1,263 @@
+#include "Tournament.h"
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+
+Tournament::Tournament(int choice, std::string& path, std::string& pathGroup)
+{
+  const int EURO = 1;
+  const int WORLD = 2;
+  if (choice == EURO)
+  {
+	try
+	{
+	  addParticipants(path, m_euroParticipants);
+	  setGroup(pathGroup, m_numberEuroGroup);
+	}
+	catch (const std::exception& ex)
+	{
+	  std::cout << "Error! "<< ex.what() << std::endl;
+	}
+	
+  }
+  if (choice == WORLD)
+  {
+	try
+	{
+	  addParticipants(path, m_worldCupParticipants);
+	  setGroup(pathGroup, m_numberWorldCupGroup);
+	}
+	catch (const std::exception& ex)
+	{
+	  std::cout << "Error!" << ex.what() << std::endl;
+	}
+  }
+}
+
+void Tournament::addParticipants(std::string& path, int numberParticipants) //The function reads participants from a file.
+{
+  std::ifstream participantsList;
+  participantsList.open(path);
+  if (!participantsList.is_open() and !participantsList.good())
+	throw std::exception("Unable to add participants. File not found");
```

Comment:
To be completely clear, file might be found, but an error could've occurred when opening it, so this message might be misleading.